test/CMakeLists.txt

```diff
@@ -319,6 +319,20 @@ else ()
   add_test(NAME order_polytope_reflection COMMAND order_polytope -tc=reflection)
   add_test(NAME order_polytope_vec_mult COMMAND order_polytope -tc=vec_mult)
 
+  add_executable (matrix_sampling_test sampling_correlation_matrices_test.cpp $<TARGET_OBJECTS:test_main>)
```

Comment:
do we need to test all those 3 methods "old", "new_point", "new_matrix" ?
Please do not include redundant ones both in the library and in the test suites but rather create an example if this is needed e.g. to show that new is better than old.

Reply:
Great thank! So this is probably a bug, right? Do you propose a fix? I still do not get why you should test  the old sampling if it is wrong. Why not opening an issue with a minimal example showing that the sampling with RDHR in Spectahedra is wrong?

Reply:
I discovered the bug while trying RDHR Walk with Spectrahedron. I also included a fix in this PR: I changed the sign (adding '-') in `line_intersection` of Spectrahedron class so now RDHR Walk works for Spectrahedron.